### DISCUSSION TITLE
Display a list of the configured shortcuts/keybinds

### DISCUSSION THREAD

**Discussion by stefonarch** [labwc/labwc#2001] [Show and tell]
As it happens that I forget what I've configured and therefor never use I tried to display all active shortcuts - with a shortcut obviously :)

![immagine](https://github.com/user-attachments/assets/c1ee7139-18a2-4a70-a928-c0cc116fd591)

The script:

```
#!/bin/bash
# Display labwc shortcuts

# Define the rc.xml file
input_file="$XDG_CONFIG_HOME/labwc/rc.xml"

generate_output() {
    xmlstarlet sel -t \
        -m "//keybind" \
        -o "" -v "@key" -o " " \
        -m "action" \
        -i "not(@name='Execute')" -o " " -v "@name" -b -o " " \
        -o " " -v "command" -o " " -v "@command" \
        -b -n \
        "$input_file"
}

generate_output |sort | zenity --text-info --title="Labwc Keybinds" --width=600 --height=700
```

**Comment by salahoued** [comment]
I had a python script that generate a pipe-menu when I was using _openbox_ ([ob-keybindings-1.py.txt](https://github.com/user-attachments/files/16315290/ob-keybindings-1.py.txt))

But now it needs some updates

**Comment by johnraff** [comment]
I don't know if (parts of) this would be any use?
https://github.com/BunsenLabs/bunsen-pipemenus/blob/carbon/bin/bl-kb-pipemenu

**Comment by 01micko** [comment]
Probably some of it at least. I'll take a look soon. Still super busy but should settle a bit in a few weeks.

 
 

Sent: Tuesday, April 28, 2026 at 5:19 PM
